circuits/src/zk_circuits/valid_wallet_create.rs

```diff
@@ -222,11 +180,50 @@ impl<const MAX_FEES: usize> CommitVerifier for ValidWalletCreateCommitment<MAX_F
 // | Statement Type Definition |
 // -----------------------------
 
-/// The parameterization for the VALID WALLET CREATE statement
-#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
-pub struct ValidWalletCreateStatement {
-    /// The expected commitment of the newly created wallet
-    pub wallet_commitment: Scalar,
+/// The statement type for the `VALID WALLET CREATE` circuit
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ValidWalletCreateStatement<
+    const MAX_BALANCES: usize,
+    const MAX_ORDERS: usize,
+    const MAX_FEES: usize,
+> {
+    /// The commitment to the private secret shares of the wallet
+    pub private_shares_commitment: Scalar,
+    /// The public secret shares of the wallet
+    pub public_wallet_shares: WalletSecretShare<MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
+}
+
+/// The statement type for the `VALID WALLET CREATE` circuit, allocated in a constraint system
+#[derive(Clone, Debug)]
+pub struct ValidWalletCreateStatementVar<
+    const MAX_BALANCES: usize,
+    const MAX_ORDERS: usize,
+    const MAX_FEES: usize,
+> {
+    /// The commitment to the private secret shares of the wallet
+    pub private_shares_commitment: Variable,
+    /// The public secret shares of the wallet
+    pub public_wallet_shares: WalletSecretShareVar<MAX_BALANCES, MAX_ORDERS, MAX_FEES>,
+}
+
+impl<const MAX_BALANCES: usize, const MAX_ORDERS: usize, const MAX_FEES: usize> CommitPublic
+    for ValidWalletCreateStatement<MAX_BALANCES, MAX_ORDERS, MAX_FEES>
+{
+    type VarType = ValidWalletCreateStatementVar<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+    type ErrorType = (); // Does not error
+
+    fn commit_public<CS: RandomizableConstraintSystem>(
+        &self,
+        cs: &mut CS,
+    ) -> Result<Self::VarType, Self::ErrorType> {
+        let private_commitment_var = self.private_shares_commitment.commit_public(cs).unwrap();
+        let public_shares_var = self.public_wallet_shares.commit_public(cs).unwrap();
```

Comment:
Is it safe to be unwrapping here (and in the other commitment traits)? prob ok for now to get this out the door anyway

Reply:
Most of these commitment traits do not error, the error is allowed for generics, but this `unwrap` and most others are safe